File: zohodesk-portal-kb/ios/RNZohoDeskPortalKB/RNZohoDeskPortalKB.m
//
//  RNZDPortalKB.m
//  RNZDPortalKB
//
//  Created by Rajesh R on 16/09/24.
//


#import <React/RCTBridgeModule.h>


@interface RCT_EXTERN_MODULE(RNZohoDeskPortalKB, NSObject)
RCT_EXTERN_METHOD(show)
RCT_EXTERN_METHOD(showArticle:(NSString *)permalink)
RCT_EXTERN_METHOD(showCategory:(NSString *)permalink)
RCT_EXTERN_METHOD(setConfiguration:(NSDictionary *)configDictionary)
@end

File: zohodesk-portal-kb/ios/RNZohoDeskPortalKB/RNZohoDeskPortalKB.swift

import Foundation
import ZohoDeskPortalKB
import ZohoDeskPortalConfiguration
import React

@objc(RNZohoDeskPortalKB)
class RNZohoDeskPortalKB: NSObject {

  @objc
  func methodQueue() -> DispatchQueue {
    return DispatchQueue.main
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }

  @objc
  func show() {
    ZDPortalKB.show(withTitle: nil, navigationMode: .present)
  }

  @objc
  func showArticle(_ permalink: String) {
    ZDPortalKB.showArticle(withPermalink: permalink, showSideMenu: true, withTitle: nil)
  }

  @objc
  func showCategory(_ permalink: String) {
    ZDPortalKB.showCategory(withPermalink: permalink, withTitle: nil)
  }

  @objc
  func setConfiguration(_ configDictionary: [String: Any]){
    ZDPortalKBConfiguration.disableArticleLike =  configDictionary["disableArticleLike"] as? Bool ?? false
    ZDPortalKBConfiguration.disableArticleDislike =  configDictionary["disableArticleDislike"] as? Bool ?? false
    ZDPortalKBConfiguration.disableArticleDetailSearch = configDictionary["disableArticleDetailSearch"] as? Bool ?? false
  }
}
